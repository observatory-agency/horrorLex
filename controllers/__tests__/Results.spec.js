const ResultsController = require('../Results');
const Search = require('../../lib/Search');
const PublicationModel = require('../../models/Publication');

jest.mock('../../lib/Search');
jest.mock('../../models/Publication');

describe('ResultsController', () => {
  let results;
  beforeEach(() => {
    Search.mockImplementation(() => ({
      many: jest.fn(() => ([])),
    }));
    results = new ResultsController();
  });

  describe('constructor', () => {
    it('should set its templates', () => {
      expect(results.template).toMatchSnapshot();
    });
  });

  describe('get', () => {
    describe('with a result', () => {
      it('should call res.render with "post" template and the results', async () => {
        const renderMock = jest.fn();
        const resMock = { render: renderMock };
        const reqMock = { query: '123' };
        const nextMock = jest.fn();
        await results.get(reqMock, resMock, nextMock);
        expect(renderMock).toHaveBeenCalledWith(results.template.get, { results: [] });
      });
    });

    describe('without a result', () => {
      it('should call res.status with a "404"', async () => {
        const statusMock = jest.fn();
        const resMock = {};
        const reqMock = { query: '123' };
        const nextMock = jest.fn();
        resMock.status = statusMock;
        Search.mockImplementation(() => ({
          many: jest.fn(() => null),
        }));
        await results.get(reqMock, resMock, nextMock);
        expect(statusMock).toHaveBeenCalledWith(404);
      });
    });

    it('should call "next" on errors', async () => {
      const error = new Error();
      const resMock = jest.fn();
      const reqMock = { query: '123' };
      const nextMock = jest.fn();
      Search.mockImplementation(() => ({
        many: jest.fn(() => { throw error; }),
      }));
      await results.get(reqMock, resMock, nextMock);
      expect(nextMock).toHaveBeenCalledWith(error);
    });
  });
});
