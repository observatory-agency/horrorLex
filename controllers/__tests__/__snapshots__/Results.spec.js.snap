// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsController constructor should set its templates 1`] = `
Object {
  "error": "error.hbs",
  "get": "results.hbs",
}
`;
